src/parser/mod.rs

```diff
@@ -9657,16 +9683,81 @@ impl<'a> Parser<'a> {
             Some(Keyword::IAM_ROLE) => CopyLegacyOption::IamRole(self.parse_iam_role_kind()?),
             Some(Keyword::IGNOREHEADER) => {
                 let _ = self.parse_keyword(Keyword::AS);
                 let num_rows = self.parse_literal_uint()?;
                 CopyLegacyOption::IgnoreHeader(num_rows)
             }
+            Some(Keyword::JSON) => CopyLegacyOption::Json,
+            Some(Keyword::MANIFEST) => {
+                let verbose = self.parse_keyword(Keyword::VERBOSE);
+                CopyLegacyOption::Manifest { verbose }
+            }
+            Some(Keyword::MAXFILESIZE) => {
+                let _ = self.parse_keyword(Keyword::AS);
+                let size = self.parse_number_value()?.value;
+                let unit = match self.parse_one_of_keywords(&[Keyword::MB, Keyword::GB]) {
+                    Some(Keyword::MB) => Some(FileSizeUnit::MB),
+                    Some(Keyword::GB) => Some(FileSizeUnit::GB),
+                    _ => None,
+                };
+                CopyLegacyOption::MaxFileSize(FileSize { size, unit })
+            }
             Some(Keyword::NULL) => {
                 let _ = self.parse_keyword(Keyword::AS);
                 CopyLegacyOption::Null(self.parse_literal_string()?)
             }
+            Some(Keyword::PARALLEL) => {
+                let enabled = match self.parse_one_of_keywords(&[
+                    Keyword::TRUE,
+                    Keyword::FALSE,
+                    Keyword::ON,
+                    Keyword::OFF,
+                ]) {
+                    Some(Keyword::TRUE) | Some(Keyword::ON) => Some(true),
+                    Some(Keyword::FALSE) | Some(Keyword::OFF) => Some(false),
+                    _ => None,
+                };
+                CopyLegacyOption::Parallel(enabled)
+            }
+            Some(Keyword::PARQUET) => CopyLegacyOption::Parquet,
+            Some(Keyword::PARTITION) => {
+                self.expect_keyword(Keyword::BY)?;
+                let columns = self.parse_parenthesized_column_list(IsOptional::Mandatory, false)?;
+                let include = self.parse_keyword(Keyword::INCLUDE);
+                CopyLegacyOption::PartitionBy(PartitionBy { columns, include })
+            }
+            Some(Keyword::REGION) => {
+                let _ = self.parse_keyword(Keyword::AS);
+                let region = self.parse_literal_string()?;
+                CopyLegacyOption::Region(region)
+            }
+            Some(Keyword::ROWGROUPSIZE) => {
+                let _ = self.parse_keyword(Keyword::AS);
+                let size = self.parse_number_value()?.value;
+                let unit = match self.parse_one_of_keywords(&[Keyword::MB, Keyword::GB]) {
+                    Some(Keyword::MB) => Some(FileSizeUnit::MB),
+                    Some(Keyword::GB) => Some(FileSizeUnit::GB),
+                    _ => None,
+                };
+                CopyLegacyOption::RowGroupSize(FileSize { size, unit })
```

Comment:
Can we pull this out to a function like `parse_file_size_unit()` so that it can be reused?